src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/DiscontinueClaimClaimantCallbackHandlerTest.java

```diff
@@ -115,4 +119,67 @@ void shouldNotPopulateSelectedClaimant_whenClaimNot2v1() {
             assertThat(response.getData().get("selectedClaimantForDiscontinuance")).isNull();
         }
     }
+
+    @Nested
+    class MidEventCheckPermissionGrantedCallback {
+
+        private static final String PAGE_ID = "checkPermissionGranted";
+
+        @Test
+        void shouldHaveNoErrors_when2v1AndPermissionGrantedDataValid() {
+            DynamicList claimantWhoIsDiscontinuingList = DynamicList.builder()
+                .value(DynamicListElement.builder()
+                           .label("Both")
+                           .build())
+                .build();
+
+            CaseData caseData = CaseDataBuilder.builder().atStateClaimSubmitted2v1RespondentRegistered().build();
+            caseData.setClaimantWhoIsDiscontinuing(claimantWhoIsDiscontinuingList);
+
+            CallbackParams params = callbackParamsOf(caseData, MID, PAGE_ID);
+            var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+
+            assertThat(response.getErrors()).isEmpty();
+        }
+
+        @Test
+        void shouldHaveNoErrors_when2v1AndPermissionDateInFuture() {
+            DynamicList claimantWhoIsDiscontinuingList = DynamicList.builder()
+                .value(DynamicListElement.builder()
+                           .label("Both")
+                           .build())
+                .build();
+
+            CaseData caseData = CaseDataBuilder.builder().atStateClaimSubmitted2v1RespondentRegistered().build();
+            caseData.setClaimantWhoIsDiscontinuing(claimantWhoIsDiscontinuingList);
+            caseData.setIsPermissionGranted(SettleDiscontinueYesOrNoList.YES);
+            caseData.setPermissionGrantedComplex(PermissionGranted.builder()
+                                                     .permissionGrantedJudge("Test")
+                                                     .permissionGrantedDate(LocalDate.of(2070, 12, 12))
+                                                     .build());
+
+            CallbackParams params = callbackParamsOf(caseData, MID, PAGE_ID);
+            var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+
+            assertThat(response.getErrors().size()).isEqualTo(1);
```

Comment:
Should have no errors but error size is 1? Is the ERROR_MESSAGE_UNABLE_TO_DISCONTINUE message?
Also can you please assert the error messages?

Reply:
Added additional assertions